tasks/post-to-cmr/tests/cmr-test.js

```diff
@@ -118,19 +119,89 @@ test.serial('postToCMR returns SIT url when CMR_ENVIRONMENT=="SIT"', async (t) =
   }
 });
 
-test.serial('postToCMR skips CMR step if the metadata file uri is missing', async (t) => {
+test.serial('postToCMR throws an error if there is no CMR metadata file', async (t) => {
   const newPayload = t.context.payload;
 
   newPayload.input.granules = [{
     granuleId: 'some granule',
     files: [{
-      filename: `s3://${t.context.bucket}/to/file.xml`
+      filename: `s3://${t.context.bucket}/to/file.blah`
     }]
   }];
 
+  try {
+    await postToCMR(newPayload);
+  }
+  catch (error) {
+    t.true(error instanceof CMRMetaFileNotFound);
+  }
+});
+
+test.serial('postToCMR throws an error if any granule is missing a metadata file', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = {
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.blah`
+    }]
+  };
+  newPayload.input.granules.push(newGranule);
+
+  try {
+    await postToCMR(newPayload);
+  }
+  catch (error) {
+    t.true(error instanceof CMRMetaFileNotFound);
+    t.is(error.message, (`CMR Meta file not found for granule ${newGranule.granuleId}`));
+  }
+});
+
+test.serial('postToCMR continues without metadata file if there is skipMetaCheck flag', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = [{
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.blah`
+    }]
+  }];
+  newPayload.input.granules = newGranule;
+  newPayload.config.skipMetaCheck = true;
+  const granuleId = newPayload.input.granules[0].granuleId;
+
   const output = await postToCMR(newPayload);
+  t.is(output.granules[0].granuleId, granuleId);
+});
+
+test.serial('postToCMR continues with skipMetaCheck even if any granule is missing a metadata file', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = {
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.blah`
+    }]
+  };
+  newPayload.input.granules.push(newGranule);
+  newPayload.config.skipMetaCheck = true;
 
-  t.is(output.granules[0].cmrLink, undefined);
+  sinon.stub(cmrjs.CMR.prototype, 'ingestGranule').callsFake(() => ({
+    result
+  }));
+  try {
+    await aws.promiseS3Upload({
+      Bucket: t.context.bucket,
+      Key: `${newPayload.input.granules[0].granuleId}.cmr.xml`,
+      Body: fs.createReadStream('tests/data/meta.xml')
+    });
+    const output = await postToCMR(newPayload);
+    t.is(
+      output.granules[0].cmrLink,
+      `https://cmr.uat.earthdata.nasa.gov/search/granules.json?concept_id=${result['concept-id']}`
+    );
+    t.is(output.granules[1].cmrLink, undefined);
+  }
+  finally {
```

Comment:
Can we also add a catch here and do `t.fail` in the catch?

Reply:
This one also should pass. But I could add the fail in 122 "postToCMR throws an error if there is no CMR metadata file" and 140 "postToCMR throws an error if any granule is missing a metadata file"